wp/headless-wp/includes/classes/Preview/PreviewLink.php

```diff
@@ -17,6 +17,96 @@ class PreviewLink {
 	 */
 	public function register() {
 		add_filter( 'template_include', [ $this, 'handle_preview' ], 20 );
+
+		add_filter( 'post_link', [ $this, 'force_posts_drafts_to_have_permalinks' ], 10, 2 );
+		add_filter( 'post_type_link', [ $this, 'force_cpts_drafts_to_have_permalinks' ], 10, 2 );
+		add_action( 'page_link', [ $this, 'force_page_drafts_to_have_permalinks' ], 10, 2 );
```

Comment:
will this have any impact if usePostLinkForRedirect is false? should this logic be optional?

Just thinking In GUS we filter post_type_link to have a structure with taxonomy in it, we had to add a check on the empty post name to not filter the link on drafts as preview would break.

Reply:
it would be a interesting use case to test. This code uses `get_sample_permalink` which still calls `get_permalink` so it would catch the changes you would have made for cpts.

Reply:
I've reworked things a bit. I'm returning the draft permalink via a new rest field that only gets used when the settting is enabled on the framework.

Therefore it won't cause any breakages in existing code. I've also added a test for a usecase like you mentioned and as you can see it works fine as long as you default post name to _something_.

I'll also update docs with best pratices for making this change work with custom permalinks like you mentioned